chapter_convolutional-modern/batch-norm_vn.md

```diff
@@ -371,13 +371,13 @@ Also note that for the sake of convenience we did not worry about automatically
 Do not worry, the Gluon `BatchNorm` layer will care of this for us.
 -->
 
-Ta tạm để các chi tiết thuật toán sang một bên mà tập trung vào các khuôn mẫu thiết kế nền tảng cho việc lập trình. 
-Thông thường, ta định nghĩa phần toán trong một hàm riêng biệt, chẳng hạn như `batch_norm`.
-Sau đó, ta tích hợp chức năng này vào một tầng tùy chỉnh, với mã nguồn chủ yếu để giải quyết các vấn đề quản trị,
-chẳng hạn như di chuyển dữ liệu đến thiết bị phù hợp ngữ cảnh, cấp phát và khởi tạo bất kỳ biến nào được yêu cầu, theo dõi các giá trị trung bình động (của trung bình và phương sai trong trường hợp này), v.v.
-Khuôn mẫu này cho phép ta tách hoàn toàn các phép tính toán ra khỏi đoạn mã phụ trợ.
-Cũng lưu ý rằng để thuận tiện, ta không cần lo về việc tự động suy ra kích thước đầu vào ở đây, do đó chúng ta cũng không cần chỉ định số lượng đặc trưng xuyên suốt.
-Đừng lo lắng, lớp `BatchNorm` của Gluon sẽ làm điều này.
+Tạm để thuật toán sang một bên và tập trung vào khuôn mẫu thiết kế (*design pattern*) của việc lập trình. 
+Thông thường, ta lập trình phần toán trong một hàm riêng biệt, ví dụ như `batch_norm`.
+Sau đó, ta tích hợp chức năng này vào một tầng tùy chỉnh, với mã nguồn chủ yếu giải quyết các vấn đề phụ trợ,
+như di chuyển dữ liệu đến thiết bị phù hợp, cấp phát và khởi tạo biến, theo dõi các giá trị trung bình động (của trung bình và phương sai trong trường hợp này), v.v.
+Khuôn mẫu này giúp tách biệt việc tính toán khỏi các đoạn mã dài.
+Cũng lưu ý rằng để thuận tiện, khi lập trình BN từ đầu, ta không tự động suy ra kích thước đầu vào, do đó cần chỉ định số lượng đặc trưng xuyên suốt.
```

Comment:
Câu này hơi khó. Hóng cao nhân

Reply:
Câu này match với phần revise sau, có 1 đoạn so sánh BN tự lập trình với BN của Gluon. BN của Gluon dùng defered initialization, còn BN tự lập trình thì ko.
>Compared with the `BatchNorm` class, which we just defined ourselves, the `BatchNorm` class defined by the `nn` model in Gluon is easier to use.
In Gluon, we do not have to worry about `num_features` or `num_dims`.
Instead, these parameter values will be inferred automatically via delayed initialization.

Reply:
Ahhh. Vậy thì hợp lý rồi. Thanks!